lib/server/render.js

```diff
@@ -0,0 +1,132 @@
+const fs = require('fs-extra');
+const React = require('react');
+const renderToStaticMarkup = require('react-dom/server').renderToStaticMarkup;
+
+const path = require('path');
+const join = path.join;
+
+const env = require('./env.js');
+const readMetadata = require('./readMetadata.js');
+const toSlug = require('../core/toSlug.js');
+const DocsLayout = require('../core/DocsLayout.js');
+const MarkdownBlock = require('../core/MarkdownBlock.js');
+
+const CWD = process.cwd();
+const siteConfig = require(CWD + '/siteConfig.js');
+
+class Markdown {
+  // links to markdown files => links to their website htmls
+  mdToHtml(Metadata) {
+    const mdToHtml = {};
+    Object.keys(Metadata).forEach(id => {
+      const metadata = Metadata[id];
+      if (metadata.language !== 'en' || metadata.original_id) {
+        return;
+      }
+      let htmlLink =
+        siteConfig.baseUrl + metadata.permalink.replace('/next/', '/');
+      if (htmlLink.includes('/docs/en/')) {
+        htmlLink = htmlLink.replace('/docs/en/', '/docs/en/VERSION/');
+      } else {
+        htmlLink = htmlLink.replace('/docs/', '/docs/VERSION/');
+      }
+      mdToHtml[metadata.source] = htmlLink;
+    });
+    return mdToHtml;
+  }
+}
+
+class Docs {
+  // determine what file to use according to its id
+  metaToFile(metadata) {
+    const language = metadata.language;
+    if (metadata.original_id) {
+      return env.translation.enabled && metadata.language !== 'en'
+        ? join(CWD, 'translated_docs', metadata.language, metadata.source)
+        : join(CWD, 'versioned_docs', metadata.source);
+    } else {
+      return env.translation.enabled && metadata.language !== 'en'
+        ? join(CWD, 'translated_docs', metadata.language, metadata.source)
+        : join(CWD, '..', readMetadata.getDocsPath(), metadata.source);
+    }
```

Comment:
Should this be here? This method should just be for the rendering and not determining which file to use?

Reply:
Yes I had debated myself on this as well. My thoughts is that we are handling from metadata to HTML, so include determining which file is logical. So in the end put this part in here, but in separate method, should be pretty clear to maintain.